security/fileCertManager/fileCertManager.go

```diff
@@ -0,0 +1,122 @@
+package fileCertManager
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"github.com/fsnotify/fsnotify"
+	"github.com/go-ocf/kit/security"
+	"io/ioutil"
+	"strings"
+)
+
+// FileCertManager holds certificates from filesystem watched for changes
+type FileCertManager struct {
+	tlsKey []byte
+	tlsCert []byte
+	tlsKeyPair tls.Certificate
+	caAuthorities *x509.CertPool
+	watcher *fsnotify.Watcher
+	tlsKeyFileName string
+	dirPath string
+	tlsCertFileName string
+}
+
+// CloseManager ends watching certificates
+func (a *FileCertManager) CloseManager() {
+	_ = a.watcher.Close()
+}
+
+func (a *FileCertManager) watchFiles() {
+	for {
+		select {
+		// watch for events
+		case event := <-a.watcher.Events:
+			{
+				if event.Op == fsnotify.Create {
+					if strings.Contains(event.Name, a.tlsKeyFileName) {
+						a.tlsKey, _ = ioutil.ReadFile(a.dirPath + "/" + a.tlsKeyFileName)
+					}
+
+					if strings.Contains(event.Name, a.tlsCertFileName) {
+						a.tlsCert, _ = ioutil.ReadFile(a.dirPath + "/" + a.tlsCertFileName)
+					}
+				}
+
+				if event.Op == fsnotify.Remove {
+					if strings.Contains(event.Name, a.tlsKeyFileName) {
+						a.tlsKey = nil
+					}
+
+					if strings.Contains(event.Name, a.tlsCertFileName) {
+						a.tlsCert = nil
+					}
+				}
+
+
+				if a.tlsCert != nil && a.tlsKey != nil {
+					cert, ert := tls.X509KeyPair(a.tlsCert, a.tlsKey)
+					if ert == nil {
+						a.tlsKeyPair = cert
```

Comment:
this must be protected by mutex - because `a.tlsKeyPair` could be accessed by multiple goroutines

Reply:
Consider it done